### PR TITLE
fix(shapewidget): Fix and improve ShapeWidget

### PR DESCRIPTION
**Pull request by tomsuchel**
### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Context

A recent change makes the widget behavior no longer share the widget factory model.
Furthermore, LabelRepresentation is about to be removed (old-style widget), so widgets using this should use SVGLandmarkRepresentation for text rendering.

### Changes

- Fix ShapeWidget (use SVGLandmarkRepresentation instead of LabelRepresentation)
- Fix PaintWidget (use SVGLandmarkRepresentation instead of LabelRepresentation)
- Fix a bug in ShapeWidget example causing labels to not being centered
- Fix PaintWidget example by importing the module HttpDataAccessHelper
- Add a way of changing text alignment and position

- [x] Documentation and TypeScript definitions were updated to match those changes

### Results

**Behaviour of text alignment**

Possible values for horizontal alignment: 
- `start`
- `middle`
- `end`

Possible values for vertical alignement (defined in `SVGLandmarkRepresentation/Constants.js`):
- `VerticalTextAlignment.START`
- `VerticalTextAlignment.MIDDLE`
- `VerticalTextAlignment.END`

Screenshot from 2021-09-13 16-26-35

**Behaviour of text position**

Possible values (defined in `ShapeWidget/Constants.js`):
- `TextPosition.MIX`
- `TextPosition.CENTER`
- `TextPosition.MAX`

Screenshot from 2021-09-13 16-26-41

**Example:**
```javascript
// Change text aligment
scene.rectangleHandle.setTextProps({
  ...scene.rectangleHandle.getTextProps(),
  'text-anchor': 'middle',                          // horizontal alignment
  verticalAlign: VerticalTextAlignment.MIDDLE,      // vertical alignment
});
// Change text position
widgets.rectangleWidget
  .getWidgetState()
  .setTextPosition([
    TextPosition.MIN,
    TextPosition.CENTER,
    TextPosition.MAX,
  ]);
```

### Testing

- [ ] This change adds or fixes unit tests
- [x] All tests complete without errors on the following environment:
  - **vtk.js**: 19.8.0
  - **OS**: Ubuntu 20.04
  - **Browser**: Chrome 94.0.4606.71

BREAKING CHANGE: in ShapeWidget: `setLabelTextCallback` is replaced by `text` substate \
BREAKING CHANGE: in ShapeWidget: `setPixelScale` has been removed. \
It should be replaced by point handle `scale1` mixin and `scaleInPixels` \
`useHandles` has been removed. It should be replaced by `setHandleVisibility`. \
`resetAfterPointPlacement` is now false by default.
BREAKING CHANGE: RectangleWidget and EllipseWidget handles now scale up automatically.

### CODE REVIEW

File: Sources/Widgets/SVG/SVGLandmarkRepresentation/Constants.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  if (fontProperties != null && fontProperties.fontSize) {
+    if (typeof fontProperties.fontSize === 'string') {
+      if (fontProperties.fontSize.slice(-2) === 'px') {
+        return parseInt(fontProperties.fontSize, 10);
+      }
+    } else {
+      return fontProperties.fontSize;
+    }
+  }
+  return 16;
```
Comment (floryst) [inline Sources/Widgets/SVG/SVGLandmarkRepresentation/Constants.js:22]: I think having a fallback font size parameter with a default value of, say, 16 would be better than returning an arbitrary number here.

File: Sources/Widgets/Widgets3D/ShapeWidget/behavior.js [hunk 2]
```diff
@@ -325,6 +256,35 @@ export default function widgetBehavior(publicAPI, model) {
...
+    // // const screenPoint2 = model.apiSpecificRenderWindow.worldToDisplay(...point2, model.renderer);
+    // const screenPoint2 = publicAPI.computeWorldToDisplay(
+    //   model.renderer,
+    //   ...point2
+    // );
+    // const screenBounds = vtkMath.computeBoundsFromPoints(
+    //   screenPoint1,
+    //   screenPoint2,
+    //   []
+    // );
```
Comment (floryst) [inline Sources/Widgets/Widgets3D/ShapeWidget/behavior.js]: Is this commented code still needed?

**Review by floryst — COMMENTED** [review]
Looks reasonable so far. I noted a few questions and changes as comments.

File: Sources/Widgets/Widgets3D/ShapeWidget/behavior.js (hunk 2 above)
Comment (tomsuchel) [reply Sources/Widgets/Widgets3D/ShapeWidget/behavior.js]: No indeed, i removed it

**Comment by floryst** [comment]
Overall this looks good to me. @finetjul is this going to conflict with your shape widget changes?

**Comment by finetjul** [comment]
@floryst nope, that's all fine. it actually fixes the troubles I had.

File: Sources/Widgets/Widgets3D/SplineWidget/index.js [hunk 3]
```diff
@@ -89,4 +89,4 @@ export const newInstance = macro.newInstance(extend, 'vtkSplineWidget');
 
 // ----------------------------------------------------------------------------
 
-export default { newInstance, extend };
+export default { newInstance, extend, widgetBehavior };
```
Comment (finetjul) [inline Sources/Widgets/Widgets3D/SplineWidget/index.js:92]: you do not need to export this one... if you want it, you simply import the behavior.js file

File: Sources/Widgets/Widgets3D/PaintWidget/index.js [hunk 4]
```diff
@@ -231,4 +231,4 @@ export const newInstance = macro.newInstance(extend, 'vtkPaintWidget');
 
 // ----------------------------------------------------------------------------
 
-export default { newInstance, extend };
+export default { newInstance, extend, widgetBehavior };
```
Comment (finetjul) [inline Sources/Widgets/Widgets3D/PaintWidget/index.js:234]: for consistency, let's move widgetBehavior into a separate `behavior.js` file

**Comment by finetjul** [comment]
LGTM

**Pull request merged**